win32ss/user/user32/misc/exticon.c

```diff
@@ -616,6 +616,11 @@ static UINT ICO_ExtractIconExW(
             goto end;
         }
 
+#ifdef __REACTOS__
+      /* CORE-15879 fix by Thomas Faber */
+      if ((ULONG_PTR)(rootresdir + 1) > (ULONG_PTR)peimage + fsizel) goto end;
```

Comment:
You are lacking the integer overflow suggested as well by Thomas here.

Reply:
I attempted to add the integer overflow check.